edge-hub/src/Microsoft.Azure.Devices.Routing.Core/RouteResult.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Routing.Core
+{
+    using System;
+
+    public class RouteResult : IEquatable<RouteResult>
+    {
+        public RouteResult(Endpoint endpoint, uint priority, uint timeToLiveSecs)
+        {
+            this.Endpoint = endpoint;
+            this.Priority = priority;
+            this.TimeToLiveSecs = timeToLiveSecs;
+        }
+
+        public readonly Endpoint Endpoint;
+        public readonly uint Priority;
+        public readonly uint TimeToLiveSecs;
+
+        public bool Equals(RouteResult other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            // Check endpoint, priority, and TTL
+            bool areEqual =
+                this.Endpoint.Equals(other.Endpoint) &&

```

Comment:
please use string.Comparison (https://docs.microsoft.com/en-us/dotnet/api/system.stringcomparison?view=netframework-4.8); is it case sensitive or insensitive?

Reply:
I don't think this class should make any assumptions about how the Endpoint.Equals is implemented, even if it's using a string compare underneath right now.

---
In reply to: [376178099](https://github.com/Azure/iotedge/pull/2469#discussion_r376178099) [](ancestors = 376178099)